### PR TITLE
feat: send capacity example

### PR DESCRIPTION
**Pull request by BaojunCZ**

### CODE REVIEW

File: console/src/main/java/org/nervos/ckb/example/TransactionExample.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    while (fromBlockNumber <= toBlockNumber && inputsCapacities.get() < needCapacities) {
+      long currentToBlockNumber = Math.min(fromBlockNumber + 100, toBlockNumber);
+      List<CellOutputWithOutPoint> cellOutputs =
+          ckbService
+              .getCellsByLockHash(
+                  lockHash, String.valueOf(fromBlockNumber), String.valueOf(currentToBlockNumber))
+              .send()
+              .getCells();
+      if (cellOutputs != null && cellOutputs.size() > 0) {
+
```
Comment (ashchan) [inline console/src/main/java/org/nervos/ckb/example/TransactionExample.java]: 1. Please do not leave blank lines after `{`.
2. Consider inserting blank line between logic trunk of code. (e.g., insert one between Ln105 and Ln106)

File: console/src/main/java/org/nervos/ckb/example/TransactionExample.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            break;
+          }
+        }
+      }
+      fromBlockNumber = currentToBlockNumber + 1;
+    }
+    return new InputsBean(cellInputs, inputsCapacities.get());
+  }
+
+  class InputsBean {
```
Comment (ashchan) [inline console/src/main/java/org/nervos/ckb/example/TransactionExample.java]: What does `InputBean` mean?

File: console/src/main/java/org/nervos/ckb/example/TransactionExample.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  private String privateKeyHex;
+  private CKBService ckbService;
+
+  public TransactionExample(String privateKeyHex, Script inputLockScript) {
+    this.privateKeyHex = privateKeyHex;
+    this.inputLockScript = inputLockScript;
+    ckbService = CKBService.build(new HttpService(NODE_URL));
+  }
+
+  public String sendCapacity(List<ReceiverBean> receiverBeanList) {
```
Comment (duanyytop) [inline console/src/main/java/org/nervos/ckb/example/TransactionExample.java]: Maybe `throw Exception` will be more better.

File: console/src/main/java/org/nervos/ckb/example/TransactionExample.java (hunk 2 above)
Comment (duanyytop) [reply console/src/main/java/org/nervos/ckb/example/TransactionExample.java]: Just `Input` is OK

File: console/src/main/java/org/nervos/ckb/example/TransactionExample.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+      fromBlockNumber = currentToBlockNumber + 1;
+    }
+    return new InputsBean(cellInputs, inputsCapacities.get());
+  }
+
+  class InputsBean {
+    List<CellInput> inputs;
+    long capacity;
+
+    public InputsBean(List<CellInput> inputs, long capacity) {
```
Comment (duanyytop) [inline console/src/main/java/org/nervos/ckb/example/TransactionExample.java]: `capacity` data type should be BigInteger or String

File: console/src/main/java/org/nervos/ckb/example/TransactionExample.java (hunk 2 above)
Comment (BaojunCZ) [reply console/src/main/java/org/nervos/ckb/example/TransactionExample.java]: All inputs and balance sum

File: console/src/main/java/org/nervos/ckb/example/TransactionExample.java (hunk 3 above)
Comment (BaojunCZ) [reply console/src/main/java/org/nervos/ckb/example/TransactionExample.java]: ok

File: console/src/test/java/org/nervos/ckb/example/TransactionExampleTest.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  public void init() {
+    String publicKey = Sign.publicKeyFromPrivate(Numeric.toBigInt(privateKey), true).toString(16);
+    String blake160 =
+        Numeric.prependHexPrefix(Numeric.cleanHexPrefix(Hash.blake2b(publicKey)).substring(0, 40));
+    inputLockScript = new Script(codeHash, Arrays.asList(blake160));
+    transactionManager = new TransactionExample(privateKey, inputLockScript);
+  }
+
+  @Test
+  public void sendToken() {
```
Comment (duanyytop) [inline console/src/test/java/org/nervos/ckb/example/TransactionExampleTest.java]: `sendCapacity` and add test as prefix

File: console/src/test/java/org/nervos/ckb/example/TransactionExampleTest.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    try {
+      TransactionExample.ReceiverBean receiver1 =
```
Comment (duanyytop) [inline console/src/test/java/org/nervos/ckb/example/TransactionExampleTest.java]: `receive1` -> `receive`

File: console/src/test/java/org/nervos/ckb/example/TransactionExampleTest.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+      Assertions.assertNotNull(hash);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  @Test
+  public void testGenerateTx() {
+    try {
+      TransactionExample.ReceiverBean receiver1 =
```
Comment (duanyytop) [inline console/src/test/java/org/nervos/ckb/example/TransactionExampleTest.java]: use `Assertions.assertThrows` for exception

File: console/src/main/java/org/nervos/ckb/example/TransactionExample.java [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+  public String sendCapacity(List<Receiver> receiverList) throws Exception {
+    Transaction transaction = generateTx(receiverList);
+    CkbTransactionHash temp = ckbService.sendTransaction(transaction).send();
+    if (temp.result == null) {
+      System.out.println(temp.error.message);
+    }
+    return temp.getTransactionHash();
+  }
+
+  private Transaction generateTx(List<Receiver> receiverBeanList) throws Exception {
```
Comment (duanyytop) [inline console/src/main/java/org/nervos/ckb/example/TransactionExample.java]: receiverBeanList -> receivers

File: console/src/main/java/org/nervos/ckb/example/TransactionExample.java [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+  private String privateKeyHex;
+  private CKBService ckbService;
+
+  public TransactionExample(String privateKeyHex, Script inputLockScript) {
+    this.privateKeyHex = privateKeyHex;
+    this.inputLockScript = inputLockScript;
+    ckbService = CKBService.build(new HttpService(NODE_URL));
+  }
+
+  public String sendCapacity(List<Receiver> receiverList) throws Exception {
```
Comment (duanyytop) [inline console/src/main/java/org/nervos/ckb/example/TransactionExample.java]: receiverList -> receivers

File: console/src/main/java/org/nervos/ckb/example/TransactionExample.java [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+    CkbTransactionHash temp = ckbService.sendTransaction(transaction).send();
+    if (temp.result == null) {
+      System.out.println(temp.error.message);
+    }
+    return temp.getTransactionHash();
+  }
+
+  private Transaction generateTx(List<Receiver> receiverBeanList) throws Exception {
+    BigInteger needCapacities = BigInteger.ZERO;
+    for (Receiver receiverBean : receiverBeanList) {
```
Comment (duanyytop) [inline console/src/main/java/org/nervos/ckb/example/TransactionExample.java]: Please delete bean

File: console/src/main/java/org/nervos/ckb/example/TransactionExample.java [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+            break;
+          }
+        }
+      }
+      fromBlockNumber = currentToBlockNumber + 1;
+    }
+    return new CellInputsAndBalanceSum(cellInputs, new BigDecimal(inputsCapacities).toBigInteger());
+  }
+
+  class CellInputsAndBalanceSum {
```
Comment (duanyytop) [inline console/src/main/java/org/nervos/ckb/example/TransactionExample.java]: CellInputs is OK. Why name is so long?

File: console/src/test/java/org/nervos/ckb/example/TransactionExampleTest.java [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+    String publicKey = Sign.publicKeyFromPrivate(Numeric.toBigInt(privateKey), true).toString(16);
+    String blake160 =
+        Numeric.prependHexPrefix(Numeric.cleanHexPrefix(Hash.blake2b(publicKey)).substring(0, 40));
+    inputLockScript = new Script(codeHash, Arrays.asList(blake160));
+    transactionManager = new TransactionExample(privateKey, inputLockScript);
+  }
+
+  @Test
+  public void testSendCapacity() throws Exception {
+    TransactionExample.Receiver receiver1 =
```
Comment (duanyytop) [inline console/src/test/java/org/nervos/ckb/example/TransactionExampleTest.java]: receive1 -> receive

**Comment by duanyytop** [comment]
Please global search and edit.

File: console/src/main/java/org/nervos/ckb/example/TransactionExample.java (hunk 11 above)
Comment (BaojunCZ) [reply console/src/main/java/org/nervos/ckb/example/TransactionExample.java]: There is no balance in CellInput,so 'CellInputs' can not describe this class. CellInputs + Balance

Comment (duanyytop) [reply console/src/main/java/org/nervos/ckb/example/TransactionExample.java]: No need to describe all inforamation

File: console/src/main/java/org/nervos/ckb/example/TransactionExample.java [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
+
+  public TransactionExample(String privateKeyHex, Script inputLockScript) {
+    this.privateKeyHex = privateKeyHex;
+    this.inputLockScript = inputLockScript;
+    ckbService = CKBService.build(new HttpService(NODE_URL));
+  }
+
+  public String sendCapacity(List<Receiver> receiverList) throws Exception {
+    Transaction transaction = generateTx(receiverList);
+    CkbTransactionHash temp = ckbService.sendTransaction(transaction).send();
```
Comment (duanyytop) [inline console/src/main/java/org/nervos/ckb/example/TransactionExample.java]: temp is too curtly and please give a meaningful name

File: console/src/main/java/org/nervos/ckb/example/TransactionExample.java [hunk 14]
```diff
@@ -0,0 +1,145 @@
...
+    String txHash = ckbService.computeTransactionHash(transaction).send().getTransactionHash();
+    Witness witness = new Witness(Numeric.toBigInt(privateKeyHex), txHash);
+    cellInputsAndBalanceSum.inputs.forEach(
+        cellInput -> {
+          transaction.witnesses.add(witness);
+        });
+    return transaction;
+  }
+
+  private CellInputsAndBalanceSum generateInputs(String lockHash, BigInteger needCapacities)
```
Comment (duanyytop) [inline console/src/main/java/org/nervos/ckb/example/TransactionExample.java]: If you want to describe all information, method name should be generateCellInputsAndBalanceSum

**Pull request merged**